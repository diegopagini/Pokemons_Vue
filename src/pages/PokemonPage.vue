<!-- @format -->

<template>
	<div>
		<h1>¿Quién es este Pokémon?</h1>

		<PokemonPicture
			:pokemonId="151"
			:showPokemon="true"
		/>

		<PokemonOptions :options="pokemons" />
	</div>
</template>

<script>
	import PokemonOptions from '@/components/PokemonOptions';
	import PokemonPicture from '@/components/PokemonPicture';

	import getPokemonOptions from '@/helpers/getPokemonOptions';

	export default {
		components: {
			PokemonOptions,
			PokemonPicture,
		},
		data() {
			return {
				pokemons: [],
			};
		},
		methods: {
			async mixPokemonArray() {
				this.pokemons = await getPokemonOptions();
			},
		},
		mounted() {
			this.mixPokemonArray();
		},
	};
</script>
