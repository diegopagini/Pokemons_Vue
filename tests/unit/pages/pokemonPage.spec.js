/** @format */
import PokemonPage from '@/pages/PokemonPage';
import { mount, shallowMount } from '@vue/test-utils';

import { POKEMONS_MOCK } from '../mocks/pokemons';

describe('PokemonPage Component', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallowMount(PokemonPage);
	});

	test('debe de hacer match con el spanshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('debe de llamar mixPokemonArray al montar', () => {
		const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray');
		const wrapper = shallowMount(PokemonPage); // To launch the mount method again.
		expect(mixPokemonArraySpy).toHaveBeenCalled();
		expect(mixPokemonArraySpy).toBeCalledTimes(1);
	});

	test('debe de hacer match con el spanshot cuando cargan los pokemons', () => {
		const wrapper = mount(PokemonPage, {
			data() {
				return {
					pokemons: POKEMONS_MOCK,
					pokemon: POKEMONS_MOCK.at(0),
					showPokemon: false,
					showAnswer: false,
					message: '',
				};
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
