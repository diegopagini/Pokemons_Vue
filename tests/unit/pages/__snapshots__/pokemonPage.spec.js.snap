// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component debe de hacer match con el spanshot 1`] = `<h1>Espere por favor...</h1>`;

exports[`PokemonPage Component debe de hacer match con el spanshot cuando cargan los pokemons 1`] = `
<div>
  <h1>¿Quién es este Pokémon?</h1>
  <div class="pokemon-container">
    <div class="pokemon-wrapper"><img alt="pokemon" class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg">
      <!--v-if-->
    </div>
  </div>
  <div class="options-container">
    <ul class="fade-in">
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
